### PR TITLE
182487074 earn milestone points

### PR DESCRIPTION
**Pull request by androidmaven**

### CODE REVIEW

File: app/src/main/java/com/hover/stax/domain/use_case/auth/AuthUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.hover.stax.domain.repository.AuthRepository
+import com.hover.stax.domain.repository.StaxUserRepository
+import com.hover.stax.utils.TokenUtils
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import timber.log.Timber
+
+class AuthUseCase(private val authRepository: AuthRepository, private val staxUserRepository: StaxUserRepository) {
+
+    fun authorize(idToken: String): Flow<Resource<StaxUser>> = flow {
```
Comment (jumaallan) [inline app/src/main/java/com/hover/stax/domain/use_case/auth/AuthUseCase.kt:14]: My initial suggestion would eliminate doing this implementation on the UseCase

**Review by jumaallan — APPROVED** [review]
Left some comments. I would like to open a PR that addresses this

File: app/src/main/java/com/hover/stax/data/remote/dto/authorization/TokenResponse.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    val accessToken: String,
+
+    @field:SerializedName("refresh_token")
+    val refreshToken: String,
+
+    @field:SerializedName("scope")
+    val scope: String,
+
+    @field:SerializedName("created_at")
+    val createdAt: Int,
```
[the remark this answers is not in the document]
Comment (davkutalek) [reply app/src/main/java/com/hover/stax/data/remote/dto/authorization/TokenResponse.kt:18]: I think it was just for simplicity with what the backend sends which is an epoch

File: app/src/main/java/com/hover/stax/domain/model/TokenInfo.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    @PrimaryKey
+    val id: Int = 1,
+
+    val accessToken: String,
+
+    val refreshToken: String,
+
+    val scope: String,
+
+    val createdAt: Int,
```
[the remark this answers is not in the document]
Comment (davkutalek) [reply app/src/main/java/com/hover/stax/domain/model/TokenInfo.kt:20]: we usually use unix epochs cause it is more universal. Rails and java dates are so different that converting is a pain and you have to worry about time zones with Java. Using a epoch avoids a lot of that hassle.

File: app/src/main/java/com/hover/stax/domain/use_case/auth/AuthUseCase.kt (hunk 1 above)
Comment (davkutalek) [reply app/src/main/java/com/hover/stax/domain/use_case/auth/AuthUseCase.kt:14]: Yeah, I just noticed how strange this is... You would put it in the repository?

**Comment by davkutalek** [comment]
@jumaallan sounds good. Please coordinate with @samuelralak for making sure development is set up correctly for testing. I believe that this change https://github.com/UseHover/server/pull/155/files needs to be reverted for it to work correctly, but there may be more to it.

**Comment by jumaallan** [comment]
> @jumaallan sounds good. Please coordinate with @samuelralak for making sure development is set up correctly for testing. I believe that this change https://github.com/UseHover/server/pull/155/files needs to be reverted for it to work correctly, but there may be more to it.

Will do. Already working on some implementation, then I will reach out to him during testing

File: app/src/main/java/com/hover/stax/domain/model/TokenInfo.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (jumaallan) [reply app/src/main/java/com/hover/stax/domain/model/TokenInfo.kt:20]: Gotcha

File: app/src/main/java/com/hover/stax/data/remote/dto/authorization/TokenResponse.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (jumaallan) [reply app/src/main/java/com/hover/stax/data/remote/dto/authorization/TokenResponse.kt:18]: Gotcha

**Comment by jumaallan** [comment]
Closing in favour of #801

**Pull request closed without merging**